nomad/structs/structs.go

```diff
@@ -12316,6 +12388,15 @@ func (a *ACLAuthMethod) Copy() *ACLAuthMethod {
 	return c
 }
 
+func (a *ACLAuthMethod) Validate() error {
+	var mErr multierror.Error
+
+	// TODO what are validity conditions for an auth method?
+	// ¯\_(ツ)_/¯
```

Comment:
😄 

Lets chat about this synchronously; but we probably want this to check all non-internally set fields have values and ensure they are not too big.

Reply:
I added a simple name validation based on the same regexp policy names are validated against, and wrote a TODO note to revisit this in the future.